packages/xdl/src/Project.js

```diff
@@ -1865,34 +1865,179 @@ function getWebpackInstance() {
   return webpackDevServerInstance;
 }
 
-async function startWebpackServerAsync(projectRoot, options, verbose) {
-  if (webpackDevServerInstance) {
-    ProjectUtils.logError(projectRoot, 'expo', 'Webpack is already running.');
-    return;
+const appDirectory = fs.realpathSync(process.cwd());
+const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
+const paths = {
+  yarnLockFile: resolveApp('yarn.lock'),
+  appTsConfig: resolveApp('tsconfig.json'),
+  appPackageJson: resolveApp('package.json'),
+};
+
+const devSocket = {
+  warnings: warnings => ProjectUtils.logInfo(projectRoot, 'warnings', warnings),
+  errors: errors => ProjectUtils.logInfo(projectRoot, 'errors', errors),
+};
+const useYarn = fs.existsSync(paths.yarnLockFile);
+const appName = require(paths.appPackageJson).name;
+
+import address from 'address';
+import formatWebpackMessages from 'react-dev-utils/formatWebpackMessages';
+import clearConsole from 'react-dev-utils/clearConsole';
+import { prepareUrls } from 'react-dev-utils/WebpackDevServerUtils';
+
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+('use strict');
+
+const isInteractive = true; //process.stdout.isTTY;
+
+function printInstructions(appName, urls, useYarn) {
+  console.log();
+  console.log(`You can now view ${chalk.bold(appName)} in the browser.`);
+  console.log();
+
+  if (urls.lanUrlForTerminal) {
+    console.log(`  ${chalk.bold('Local:')}            ${urls.localUrlForTerminal}`);
+    console.log(`  ${chalk.bold('On Your Network:')}  ${chalk.underline(urls.lanUrlForTerminal)}`);
+  } else {
+    console.log(`  ${urls.localUrlForTerminal}`);
   }
+
+  console.log();
+  console.log('Note that the development build is not optimized.');
+  console.log(`To create a production build, use ` + `${chalk.cyan(`expo build:web`)}.`);
+  console.log();
+}
+
+function createCompiler({ appName, config, devSocket, urls, useYarn, webpack, onFinished }) {
+  // "Compiler" is a low-level interface to Webpack.
+  // It lets us listen to some events and provide our own custom messages.
+  let compiler;
+  try {
+    compiler = webpack(config);
+  } catch (err) {
+    console.log(chalk.red('Failed to compile.'));
+    console.log();
+    console.log(err.message || err);
+    console.log();
+    process.exit(1);
+  }
+
+  // "invalid" event fires when you have changed a file, and Webpack is
+  // recompiling a bundle. WebpackDevServer takes care to pause serving the
+  // bundle, so if you refresh, it'll wait instead of serving the old one.
+  // "invalid" is short for "bundle invalidated", it doesn't imply any errors.
+  compiler.hooks.invalid.tap('invalid', () => {
+    if (isInteractive) {
+      clearConsole();
+    }
+    console.log(chalk.white('Compiling...'));
```

Comment:
Can you route these logs through the log functions in `ProjectUtils` (`ProjectUtils.logInfo` etc) so that they work correctly with Expo Dev Tools, debug logging etc.?